src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/persistence/SqliteObjectPersistor.java

```diff
@@ -0,0 +1,309 @@
+
+    class NoGetter {
+      @ValueColumn
+      int x;
+    }
+
+    SQLitePersistor sqlite = new SQLitePersistor(
+        testLocation.toString(), baseFilename, String.valueOf(1), TimeUnit.SECONDS, 1);
+    sqlite.write(new NoGetter());
+  }
+
+  @Test
+  public void testGetterReturnMismatch()
+      throws IOException, SQLException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+    exceptionRule.expect(NoSuchMethodException.class);
+    exceptionRule.expectMessage("The return type of the getter 'getX' (class java.lang.Integer) and field 'x' (int) don't match.");
+    class TypeMismatch {
+      @ValueColumn
+      int x;
+
+      public Integer getX() {
+        return x;
+      }
+
+      public void setX(int x) {
+        this.x = x;
+      }
+    }
+
+    SQLitePersistor sqlite = new SQLitePersistor(
+        testLocation.toString(), baseFilename, String.valueOf(1), TimeUnit.SECONDS, 1);
+    sqlite.write(new TypeMismatch());
+  }
+
+  @Test
+  public void testSetterArgTypeMismatch()
+      throws IOException, SQLException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+    exceptionRule.expect(NoSuchMethodException.class);
+    exceptionRule.expectMessage("Could not find 'setter' for the field 'x' of class");
+    class TypeMismatch {
+      @ValueColumn
+      int x;
+
+      public int getX() {
+        return x;
+      }
+
+      public void setX(Integer x) {
+        this.x = x;
+      }
+    }
+
+    SQLitePersistor sqlite = new SQLitePersistor(
+        testLocation.toString(), baseFilename, String.valueOf(1), TimeUnit.SECONDS, 1);
+    sqlite.write(new TypeMismatch());
+  }
+
+  @Test
+  public void testNonPublicGetterSetter()
+      throws IOException, SQLException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+    exceptionRule.expect(IllegalStateException.class);
+    exceptionRule.expectMessage("Found 'getX'. But it is not public");
+    class TypeMismatch {
+      @ValueColumn
+      int x;
+
+      int getX() {
+        return x;
+      }
+
+      public void setX(Integer x) {
+        this.x = x;
+      }
+    }
+
+    SQLitePersistor sqlite = new SQLitePersistor(
+        testLocation.toString(), baseFilename, String.valueOf(1), TimeUnit.SECONDS, 1);
+    sqlite.write(new TypeMismatch());
+  }
+
+  @Test
+  public void testNoPersistableFields()
+      throws IOException, SQLException, IllegalAccessException, NoSuchMethodException, InvocationTargetException {
+    exceptionRule.expect(IllegalStateException.class);
+    exceptionRule.expectMessage(
+        "NotPersistable was asked to be persisted but there are no fields with annotations: ValueColumn or RefColumn");
+    class NotPersistable {
+      int x;
+    }
+    
+    SQLitePersistor sqlite = new SQLitePersistor(
+        testLocation.toString(), baseFilename, String.valueOf(1), TimeUnit.SECONDS, 1);
+    sqlite.write(new NotPersistable());
+  }
+
+  static class Outer {
```

Comment:
Can we rename this class as say PersistorTestExample or MockPersistorInput?

Reply:
done